lib/tapioca/helpers/rbi_helper.rb

```diff
@@ -28,10 +35,60 @@ def update_gem_rbis_strictnesses(gem_names, gem_dir: DEFAULT_GEM_DIR, dsl_dir: D
       errors = Spoom::Sorbet::Errors::Parser.parse_string(res.err)
 
       if errors.empty?
-        say("No error found", [:green, :bold])
+        say("  No error found\n\n", [:green, :bold])
         return
       end
 
+      parse_errors = errors.select { |error| error.code < 4000 }
+
+      if parse_errors.any?
+        say_error(<<~ERR, :red)
+
+          ##### INTERNAL ERROR #####
+
+          There are parse errors in the generated RBI files.
+
+          This seems related to a bug in Tapioca.
+          Please open an issue at https://github.com/Shopify/tapioca/issues/new with the following information:
+
+          Tapioca v#{Tapioca::VERSION}
+
+          Command:
+            #{command}
+
+        ERR
+
+        say_error(<<~ERR, :red) if gems.any?
+          Gems:
+          #{gems.map { |gem| "  #{gem.name} (#{gem.version})" }.join("\n")}
+
+        ERR
+
+        say_error(<<~ERR, :red) if compilers.any?
+          Compilers:
+          #{compilers.map { |compiler| "  #{compiler.name}" }.join("\n")}
+
+        ERR
+
+        say_error(<<~ERR, :red)
+          Errors:
+          #{parse_errors.map { |error| "  #{error}" }.join("\n")}
+
+          ##########################
+
+        ERR
+      end
+
+      redef_errors = errors.select { |error| error.code == 4010 }
+      update_gem_rbis_strictnesses(redef_errors, gems.map(&:name), gem_dir) if auto_strictness
```

Comment:
I think the following could be clearer:
```suggestion
      if auto_strictness
        redef_errors = errors.select { |error| error.code == 4010 }
        update_gem_rbis_strictnesses(redef_errors, gems.map(&:name), gem_dir) 
      end
```